sif/sif/sif.go

```diff
@@ -0,0 +1,297 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// Copyright (c) 2017, SingularityWare, LLC. All rights reserved.
+// Copyright (c) 2017, Yannick Cote <yhcote@gmail.com> All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+// +build linux
```

Comment:
Just removing these build conditionals, the code seems to at least build fine on macOS.

Of course things like `unsquashfs` or `fakeroot` might not be available — but the error handling needs to be user-acceptable for a Linux only version as well.  At least from a very quick search, both seem to be packaged in Homebrew.

Reply:
`fakeroot` does exist in homebrew, but it installs with
> Warning: fakeroot has been deprecated because it does not build!

and it also doesn’t work — in wildly unpredictable ways (sometimes files are not created, sometimes chown fails with EPERM, sometimes chown doesn’t fail but doesn’t do anything).

So making this Linux-only seems reasonable for now. (Ideally we shouldn’t need `fakeroot`, and do the conversion from SquashFS to tar most in-memory.)